fe/fe-core/src/main/java/org/apache/doris/dictionary/DictionaryManager.java

```diff
@@ -0,0 +1,365 @@
+    }
+
+    /**
+     * Drop all dictionaries in a table. Used when dropping a table. So maybe no db or table records.
+     */
+    public void dropTableDictionaries(String dbName, String tableName) {
+        lockWrite();
+        List<Dictionary> droppedDictionaries = Lists.newArrayList();
+        try {
+            ListMultimap<String, String> tableToDicNames = dbTableToDicNames.get(dbName);
+            if (tableToDicNames == null) { // this db has no table with dictionary records.
+                return;
+            }
+            // get all dictionary names of this table
+            List<String> dictNames = tableToDicNames.removeAll(tableName);
+            if (dictNames == null) { // this table has no dictionaries.
+                return;
+            }
+            // all this db's dictionaries. tableToDicNames is not null so nameToDics must not be null.
+            Map<String, Dictionary> nameToDics = dictionaries.get(dbName);
+            for (String dictName : dictNames) {
+                Dictionary dictionary = nameToDics.remove(dictName);
+                droppedDictionaries.add(dictionary);
+                // Log the drop operation
+                Env.getCurrentEnv().getEditLog().logDropDictionary(dbName, dictName);
+            }
+        } finally {
+            unlockWrite();
+        }
+        for (Dictionary dictionary : droppedDictionaries) {
+            scheduleDataUnload(null, dictionary);
+        }
+    }
+
+    /**
+     * Drop all dictionaries in a database. Used when dropping a database.
+     */
+    public void dropDbDictionaries(String dbName) {
+        lockWrite();
+        List<Dictionary> droppedDictionaries = Lists.newArrayList();
+        try {
+            // pop and save item from dictionaries
+            Map<String, Dictionary> dbDictionaries = dictionaries.remove(dbName);
+            // Log the drop operation
+            if (dbDictionaries != null) {
+                for (Map.Entry<String, Dictionary> entry : dbDictionaries.entrySet()) {
+                    droppedDictionaries.add(entry.getValue());
+                    Env.getCurrentEnv().getEditLog().logDropDictionary(dbName, entry.getKey());
+                }
+                // also drop all name mapping records.
+                dbTableToDicNames.remove(dbName);
+            }
+        } finally {
+            unlockWrite();
+        }
+        for (Dictionary dictionary : droppedDictionaries) {
+            scheduleDataUnload(null, dictionary);
+        }
+    }
+
+    /**
+     * Check if a dictionary exists.
+     */
+    public boolean hasDictionary(String dbName, String dictName) {
+        lockRead();
+        try {
+            Map<String, Dictionary> dbDictionaries = dictionaries.get(dbName);
+            return dbDictionaries != null && dbDictionaries.containsKey(dictName);
+        } finally {
+            unlockRead();
+        }
+    }
+
+    public Map<String, Dictionary> getDictionaries(String dbName) {
```

Comment:
the method ater call the func，the map not protected by lock

Reply:
This function only used in show command. I think it only requies consistency but not real time

Reply:
This function is only used in the show command. It returns a ConcurrentHashMap with weak consistency, which I think is enough for the show command.

Reply:
Do a copy here not do the dirty work！